### PR TITLE
Revert "Only skip password types for now"

### PR DESCRIPTION
**Pull request by alexjh**
This reverts commit df3f26631e2b10ed5b51d4289df99b35a402b7ac.

This allows `scf-secret-generator` to update certs and SSH keys as well.

### CODE REVIEW

File: cmd/build-kube.go [hunk 1]
```diff
@@ -97,5 +100,12 @@ func init() {
 		"Additional information to use in computing the image tags",
 	)
 
+	buildKubeCmd.PersistentFlags().BoolP(
+		"use-secrets-generator",
+		"",
+		false,
+		"The project uses the a secret generator pre-flight job",
```
Comment (alexjh) [inline cmd/build-kube.go]: `s/the a/the` ?

File: cmd/build-helm.go [hunk 2]
```diff
@@ -97,5 +100,12 @@ func init() {
 		"Additional information to use in computing the image tags",
 	)
 
+	buildHelmCmd.PersistentFlags().BoolP(
+		"use-secrets-generator",
+		"",
+		false,
+		"The project uses the a secret generator pre-flight job",
```
Comment (alexjh) [inline cmd/build-helm.go]: `s/the a/the`

File: cmd/build-kube.go (hunk 1 above)
Comment (alexjh) [inline cmd/build-kube.go]: `s/the a/the`

File: kube/stateful_set.go [hunk 3]
```diff
@@ -30,6 +30,12 @@ func NewStatefulSet(role *model.Role, settings ExportSettings) (helm.Node, helm.
 	spec := helm.NewMapping()
 	spec.Add("serviceName", fmt.Sprintf("%s-set", role.Name))
 	spec.Add("template", podTemplate)
+	// "updateStrategy" is new in kube 1.7, so we don't add anything to non-helm configs
+	// The default behaviour is "OnDelete"
```
Comment (mook-as) [inline kube/stateful_set.go]: This feels like it belongs in a different PR (it's not directly related to secret generation)

Comment (jandubois) [reply kube/stateful_set.go]: I agree; it was only merged in here because this integration branch was a substitute for `master` until we had restored backwards compatibility. Do you think this is important enough to split out into a separate PR. It is already in a separate commit: https://github.com/SUSE/fissile/pull/301/commits/4a2a4f866640c4d5d7d5d8fe96be86a4ad5d2b68, which points back to its own Trello card...

File: cmd/build-helm.go [hunk 4]
```diff
@@ -97,5 +100,12 @@ func init() {
 		"Additional information to use in computing the image tags",
 	)
 
+	buildHelmCmd.PersistentFlags().BoolP(
+		"use-secrets-generator",
+		"",
+		false,
+		"The project uses a secret generator pre-flight job",
```
[the remark this answers is not in the document]
Comment (jandubois) [reply cmd/build-helm.go]: I have problems creating a short description what it does beyond "preserve the old functionality before we had secrets generator roles".

Some of the things it does:

* no longer generates passwords in the kube secrets
* doesn't include default settings for generated secrets other than passwords (SSH keys and SSL certs and CAs) in `values.yaml`
* calls the secret object `secrets-update` instead of `secrets` but still references the `secret` object from all other objects

I would rather document these requirements in the `README.md` of the `fissile-secrets-generator` repo. I think it is too much detail for a CLI option summary.

But anyways, what would be a better name for this option?

[the remark this answers is not in the document]
Comment (jandubois) [reply cmd/build-helm.go]: > doesn't include default settings for generated secrets other than passwords

Ok, I can barely understand what I wrote myself; what I meant was:

* includes the names of generated secrets in `values.yaml`, but ignores the values provided by `--defaults-file`
* this includes the names of generated password, which aren't user-overridable without this option

[the remark this answers is not in the document]
Comment (mook-as) [reply cmd/build-helm.go]: Best name I can think of is `--skip-generated-secrets`, but that's not _really_ better.

I think I'm okay with leaving it as `--use-secrets-generator` if the help text explains how to use it (or point to a place that tells you how to use it, given the verbosity).

File: kube/stateful_set.go (hunk 3 above)
Comment (mook-as) [reply kube/stateful_set.go]: 😑 Alright, I guess.
